src/Development/IDE/Core/Rules.hs

```diff
@@ -506,6 +506,57 @@ getHiFileRule = defineEarlyCutoff $ \GetHiFile f -> do
                     <> errMsg
                   return (Nothing, ([d], Nothing))
 
+failRule :: IdeResult a
+failRule = ([], Nothing)
+
+logStr :: String -> Action ()
+logStr t = do
+    logger <- actionLogger
+    liftIO $ logDebug logger $ T.pack t
+
+getModSummaryRule :: Rules ()
+getModSummaryRule = define $ \GetModSummary f -> do
+    session <- hscEnv <$> use_ GhcSession f
+    let dflags = hsc_dflags session
+        filePath = fromNormalizedFilePath f
+        fileName = takeFileName filePath
+    (_, mFileContent) <- getFileContents f
+    fileContent <- liftIO $ maybe (hGetStringBuffer filePath) return mFileContent
```

Comment:
why call `hGetStringBuffer`? I would use `failRule` here.

EDIT: right, `getFileContents` only returns for files of interest. Fair enough, the rule for `ParsedModule` does the same.